src/compiler/checker.ts

```diff
@@ -14230,7 +14230,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function getConstraintFromIndexedAccess(type: IndexedAccessType) {
-        if (isMappedTypeGenericIndexedAccess(type) || isGenericMappedType(type.objectType)) {

```

Comment:
while my comment [here](https://github.com/microsoft/TypeScript/pull/53066#discussion_r1123593187) can be partially true - so far I couldn't find the situation in which this extra check would benefit anything (after adjustments) since `getSimplifiedIndexedAccessType` already performs this substitution when it's safe to do so